acceptance/tests/api-gateway/api_gateway_kitchen_sink_test.go

```diff
@@ -0,0 +1,282 @@
+		"externalServers.httpsPort":                   "8501",
+		"global.tls.caCert.secretName":                fmt.Sprintf("%s-consul-ca-cert", serverReleaseName),
+		"global.tls.caCert.secretKey":                 "tls.crt",
+		"global.acls.bootstrapToken.secretName":       fmt.Sprintf("%s-consul-bootstrap-acl-token", serverReleaseName),
+		"global.acls.bootstrapToken.secretKey":        "token",
+	}
+
+	releaseName := helpers.RandomName()
+	consulCluster := consul.NewHelmCluster(t, helmValues, ctx, cfg, releaseName)
+	consulCluster.SkipCheckForPreviousInstallations = true
+
+	consulCluster.Create(t)
+
+	// Override the default proxy config settings for this test
+	consulClient, _ := consulCluster.SetupConsulClient(t, true, serverReleaseName)
+	logger.Log(t, "have consul client")
+	_, _, err := consulClient.ConfigEntries().Set(&api.ProxyConfigEntry{
+		Kind: api.ProxyDefaults,
+		Name: api.ProxyConfigGlobal,
+		Config: map[string]interface{}{
+			"protocol": "http",
+		},
+	}, nil)
+	require.NoError(t, err)
+	logger.Log(t, "set consul config entry")
+
+	logger.Log(t, "creating other namespace")
+	out, err := k8s.RunKubectlAndGetOutputE(t, ctx.KubectlOptions(t), "create", "namespace", "other")
+	require.NoError(t, err, out)
+	helpers.Cleanup(t, cfg.NoCleanupOnFailure, cfg.NoCleanup, func() {
+		// Ignore errors here because if the test ran as expected
+		// the custom resources will have been deleted.
+		k8s.RunKubectlAndGetOutputE(t, ctx.KubectlOptions(t), "delete", "namespace", "other")
+	})
+
+	// create a GatewayClassConfig with configuration set
+	gatewayClassConfigName := "kitchen-sink-gateway-class-config"
+	gatewayClassName := "kitchen-sink"
+	gatewayClassConfig := &v1alpha1.GatewayClassConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: gatewayClassConfigName,
+		},
+		//Set up GatewayClassConfig specific features
+		Spec: v1alpha1.GatewayClassConfigSpec{
+			DeploymentSpec: v1alpha1.DeploymentSpec{
+				DefaultInstances: pointer.Int32(2),
+				MaxInstances:     pointer.Int32(3),
+				MinInstances:     pointer.Int32(1),
+			},
+		},
+	}
+	k8sClient := ctx.ControllerRuntimeClient(t)
+
+	logger.Log(t, "creating gateway class config")
+	err = k8sClient.Create(context.Background(), gatewayClassConfig)
+	require.NoError(t, err)
+	helpers.Cleanup(t, cfg.NoCleanupOnFailure, cfg.NoCleanup, func() {
+		logger.Log(t, "deleting all gateway class configs")
+		k8sClient.DeleteAllOf(context.Background(), &v1alpha1.GatewayClassConfig{})
+	})
+
+	gatewayParametersRef := &gwv1beta1.ParametersReference{
+		Group: gwv1beta1.Group(v1alpha1.ConsulHashicorpGroup),
+		Kind:  gwv1beta1.Kind(v1alpha1.GatewayClassConfigKind),
+		Name:  gatewayClassConfigName,
+	}
+
+	// Create gateway class referencing gateway-class-config.
+	logger.Log(t, "creating controlled gateway class")
+	createGatewayClass(t, k8sClient, gatewayClassName, gatewayClassControllerName, gatewayParametersRef)
```

Comment:
do we still need to create the gatewayclass and all that manually?

Reply:
I think `managedGatewayClass` defaults to true so we should be able to skip these steps if there isn't anything special about the `classConfig`.

Reply:
I could swap it over to kustomize before I merge it in for sure. I can't remember why it was split out this way.

Reply:
`kitchen-sink-gateway-class-config` does have configuration that utilities the min/max deployment config.